### PR TITLE
add payload CORS config

### LINKED ISSUE #323: @/charterafrica CORS error
**Describe the bug**
Access to fetch at 'https://charterafrica.prod.codeforafrica.org/api' from origin 'https://charter.africa'  being blocked by CORS

### PR DESCRIPTION
**Pull request by thepsalmist**
## Description

Add CORS configuration to allow request from charter.africa domain

Fixes #323 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: apps/charterafrica/.env.template [hunk 1]
```diff
@@ -4,6 +4,7 @@ MONGO_URL
 
 PAYLOAD_PUBLIC_APP_URL
 PAYLOAD_SECRET_KEY
+PAYLOAD_ALLOWED_DOMAINS=
```
Comment (kilemensi) [inline apps/charterafrica/.env.template:7]:
```suggestion
PAYLOAD_CORS=
```
1. This is not for domains but full URLs.
2. If it needs to be accessible to the front-end (similar to `APP_URL`), it must have PUBLIC prefix e.g. `PAYLOAD_PUBLIC_CORS`.
3. Order env vars alphabetically.

File: apps/charterafrica/payload.config.js [hunk 2]
```diff
@@ -73,6 +73,7 @@ export default buildConfig({
       },
     }),
   },
+  cors: [process.env.PAYLOAD_ALLOWED_DOMAINS.split(",")],
```
Comment (kilemensi) [inline apps/charterafrica/payload.config.js:76]:
```suggestion
  cors,
```

... and do the processing just after [`const appURL = ...`](https://github.com/CodeForAfrica/ui/pull/325/files#diff-1b3a74b7cd022ec463c5322d5a323ff7d8624f32827817b84748c4401568844dR32)
```js
const cors = process?.env?.PAYLOAD_CORS?.split(",")
  ?.map((d) => d.trim())
  ?.filter(Boolean)
```

We want `cors` to be `undefined` if no URLs were set. We don't want it to be `[undefined]`.

File: turbo.json [hunk 3]
```diff
@@ -30,7 +30,8 @@
         "S3_BUCKET",
         "S3_SECRET_ACCESS_KEY",
         "S3_REGION",
-        "SENDGRID_API_KEY"
+        "SENDGRID_API_KEY",
+        "PAYLOAD_ALLOWED_DOMAINS"
```
Comment (kilemensi) [inline turbo.json:34]: Order these alphabetically.

**Review by kilemensi — COMMENTED** [review]
👍🏽

File: turbo.json (hunk 3 above)
Comment (thepsalmist) [reply turbo.json:34]: Sure

**Review by kilemensi — APPROVED** [review]
🚀

**Pull request merged**